### PR TITLE
PLFM-7968: Refactor multi-value columns to avoid using companion table

### PR DESCRIPTION
**Pull request by marcomarasca**

### CODE REVIEW

File: lib/lib-table-cluster/src/test/java/org/sagebionetworks/table/cluster/SQLTranslatorUtilsTest.java [hunk 1]
```diff
@@ -2536,7 +2539,7 @@ public void testTranslateModelWithHasLikeKeywordAndEscape() throws ParseExceptio
 		TableAndColumnMapper mapper = new TableAndColumnMapper(element, mockSchemaProvider);
 		Map<String, Object> parameters = new HashMap<>();
 		SQLTranslatorUtils.translateModel(element, parameters, userId, mapper);
-		assertEquals( "SELECT * FROM T123 WHERE ROW_ID IN ( SELECT ROW_ID_REF_C777_ FROM T123_INDEX_C777_ WHERE _C777__UNNEST IN ( :b0, :b1, :b2 ) ) AND ( ROW_ID IN ( SELECT ROW_ID_REF_C111_ FROM T123_INDEX_C111_ WHERE _C111__UNNEST LIKE :b3 ESCAPE :b5 OR _C111__UNNEST LIKE :b4 ESCAPE :b5 ) OR _C333_ = :b6 )",element.toSql());
+		assertEquals("SELECT * FROM T123 WHERE ( JSON_OVERLAPS(_C777_,JSON_ARRAY(:b0,:b1,:b2)) IS TRUE ) AND ( ( JSON_SEARCH(_C111_,'one',:b3 COLLATE 'utf8mb4_0900_ai_ci',:b5,'$[*]') IS NOT NULL OR JSON_SEARCH(_C111_,'one',:b4 COLLATE 'utf8mb4_0900_ai_ci',:b5,'$[*]') IS NOT NULL ) OR _C333_ = :b6 )",element.toSql());
 		assertEquals(1L, parameters.get("b0"));
```
Comment (john-hill) [inline lib/lib-table-cluster/src/test/java/org/sagebionetworks/table/cluster/SQLTranslatorUtilsTest.java:2571]: need a test for the case where the 'not has_like' -> is null

File: lib/lib-table-cluster/src/main/java/org/sagebionetworks/table/cluster/SQLUtils.java [hunk 2]
```diff
@@ -1163,8 +1163,15 @@ private static void appendAddIndex(StringBuilder builder, DatabaseColumnInfo inf
 	 * @param columnId
 	 * @return
 	 */
-	public static String getIndexName(String columnId){
-		return columnId+IDX;
+	public static String getIndexName(String columnName){
+		return columnName+IDX;
+	}
+	
+	public static String getColumnNameFromIndex(String indexName) {
```
Comment (john-hill) [inline lib/lib-table-cluster/src/main/java/org/sagebionetworks/table/cluster/SQLUtils.java:1170]: optional?

File: lib/lib-table-cluster/src/test/java/org/sagebionetworks/table/cluster/TableIndexDAOImplTest.java [hunk 3]
```diff
@@ -4304,7 +4318,7 @@ public void testCopyEntityReplicationToViewWithRowFilterNull() {
 	@Test
 	public void testPopulateListColumnIndexTableView_NoIdFilter() {
 		indexDescription = new ViewIndexDescription(tableId, TableType.entityview);
-		int rowCount = 4;
+		int rowCount = 1000;
 		boolean includeMultiValue = true;
```
Comment (john-hill) [inline lib/lib-table-cluster/src/test/java/org/sagebionetworks/table/cluster/TableIndexDAOImplTest.java]: too high!

**Review by john-hill — APPROVED** [review]
Good Job!!!!!!!!!!!!!!!!!!!!!!!!

File: lib/lib-table-cluster/src/test/java/org/sagebionetworks/table/cluster/SQLTranslatorUtilsTest.java (hunk 1 above)
Comment (marcomarasca) [reply lib/lib-table-cluster/src/test/java/org/sagebionetworks/table/cluster/SQLTranslatorUtilsTest.java:2571]: I added the test and the query was indeed wrong, NOT HAS_LIKE(foo, 'a', 'b') was translated into JSON_SEARCH(foo, 'a') IS NULL **OR** JSON_SEARCH(foo, 'b') IS NULL. Should be instead JSON_SEARCH(foo, 'a') IS NULL **AND** JSON_SEARCH(foo, 'b') IS NULL
